classes/profile.php

```diff
@@ -268,17 +268,21 @@ protected function check_update_userid(int $currentid) : bool {
             return true;
         }
 
-        // If current userid isn't guest, but stored user is guest, update it.
+        if ($currentid === $stored) {
+            return false;
+        }
```

Comment:
This block is great. You've managed to simplify the following blocks by pulling out a common factor and returning a result early, and also because of that it should in most cases be quicker to evaluate which means better performance. :gem: 

Could we add a comment to this block? It would make it more consistent with the rest of the conditional blocks in this method and reinforces the intent. Something along the lines of "If the would-be user matches the currently stored, do not update".

Reply:
done